Polygon_mesh_processing/test/Polygon_mesh_processing/test_orient_cc.cpp

```diff
@@ -5,20 +5,97 @@
 #include <iostream>
 #include <fstream>
 
-using namespace CGAL;
-namespace PMP = Polygon_mesh_processing;
+namespace PMP = CGAL::Polygon_mesh_processing;
 
-typedef Exact_predicates_inexact_constructions_kernel Kernel;
-typedef Surface_mesh<Kernel::Point_3> Surface_mesh;
+typedef CGAL::Exact_predicates_inexact_constructions_kernel Kernel;
+typedef CGAL::Surface_mesh<Kernel::Point_3> SMesh;
 
+template<class TriangleMesh, class NamedParameters>
+bool test_orientation(TriangleMesh& tm, const NamedParameters& np)
+{
```

Comment:
you should not duplicate the code, it's too hard to maintain.
You can
- run `orient_connected_components()`,
- then `connected_components()` on the output,
- then iterate over the connected components, and check their orientation with `is_outward_oriented()`

Reply:
"iterate over the connected components, and check their orientation with is_outward_oriented() "
is basically 90% of what orient_connected_components does, how can I not "duplicate" ?

Reply:
you can remove everything about named parameters (just pass `np` to your functions), and everything about extreme vertices

Reply:
Extreme vertices are used to be sure is_outward_oriented() works, I am not sure it is a good idea to remove them.